packages/cyberstorm/src/components/DropDown.tsx

```diff
@@ -0,0 +1,113 @@
+import React, { ReactElement, ReactNode, useState } from "react";
+import styles from "./componentStyles/DropDown.module.css";
+import { Button } from "./Button";
+import { faChevronDown } from "@fortawesome/free-solid-svg-icons";
+import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
+
+import * as DropdownMenu from "@radix-ui/react-dropdown-menu";
+
+type DropDownProps = {
+  label?: string;
+  icon?: ReactNode;
+  defaultOpen: boolean;
+  dropDownData?: DataItem[];
+  colorScheme?: "default" | "defaultDark" | "primary";
+  defaultValue?: number;
+};
+
+type DataItem = {
+  label: string;
+  value: number;
+  reactElement?: ReactElement;
```

Comment:
Additionally, based on how this is used in the stories, and the fact that we're trying to pass `colorScheme` props to this component in `mapDropDownData`, should the actual type be `typeof DropDownItem`? That would limit the what can be passed into the dropdown, but I don't see that as a bad thing, especially since `DropDownItem` is already quite flexible in what it can do.

Reply:
Makes sense. We can extend the DropDownItem if needed,
Idea was to allow passing any elements, but we can probably add a global "CyberstormComponent" type (which'd take colorScheme as a prop) in the future if needed. 
Anyway refactored, should be a little better now.